src/coreclr/jit/morph.cpp

```diff
@@ -5603,8 +5603,6 @@ GenTree* Compiler::fgMorphField(GenTree* tree, MorphAddrContext* mac)
         tree->SetOper(GT_IND);
         tree->AsOp()->gtOp1 = addr;
 
-        tree->SetIndirExceptionFlags(this);
```

Comment:
The issue with this `SetIndirExceptionFlags` is that it is done in pre-order, before the child `FIELD`s are morphed, and those change their semantics based on the morphing context we pass down, but `fgAddrCouldBeNull` doesn't know about that (and thinks `ADDR(FIELD)` is always non-null).